File: project10/ruby/Ostrich.rb
# Ostrich.rb | Defines the Ostrich class which inherits attributes and methods
#
# Completed by: Haim Hong
# Date: March 25 2022
####################################################

require './Walkingbird.rb'

class Ostrich < Walkingbird
    def call # call function for this class specific
        'Snork!'
    end
end

File: project10/java/Ostrich.java
/* Ostrich.java provides an Ostrich subclass of Walkingbird.
 *
 * Completed by: Haim Hong
 * Date: March 25 2022
 ******************************************************/

public class Ostrich extends WalkingBird{
    /* default constructor
     * PostCond: myName == "".
     */
     public Ostrich(){
       super();
     }
    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
     public Ostrich(String name){
       super(name);
     }
   
   
    /* An Ostrich's Call
     * Return: an owl-call ("Whoo-hoo!").
     */
     public String call(){
       return "Snork!";
     }
   } 